tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -0,0 +1,150 @@
+import httpclient, asynchttpserver, asyncdispatch, asyncfutures
+import net
+
+const postBegin = """
+POST / HTTP/1.1
+Host: 127.0.0.1:64123
+Accept: */*
+Transfer-Encoding:chunked
+Content-Type: application/x-www-form-urlencoded
+
+"""
+var
+    sanitySimpleHelloWorldTest = false
+    sanitySimpleEncodingTest = false
+    sanitySimpleChunksTest = false
+    sanityComplexChunksTest = false
+
+proc testSimpleHelloWorld() =
+    let exampleWorld = ("b\r\n" &
+                        "hello=world\r\n" &
+                        "0\r\n" &
+                        "\r\n")
+    let expected = "hello=world"
+
+    proc handler(request: Request) {.async.} =
+        doAssert(request.body == expected)
+        doAssert(request.headers.hasKey("Transfer-Encoding"))
+        doAssert(not request.headers.hasKey("Content-Length"))
+        sanitySimpleHelloWorldTest = true
+        await request.respond(Http200, "Good")
+
+    let server = newAsyncHttpServer()
+    discard server.serve(Port(64123), handler)
```

Comment:
such hardcoded ports can cause flaky tests due to address already in use, see https://github.com/nim-lang/Nim/pull/14327 for context that fixed similar bugs.
(and indeed tasynchttpserver.nim uses that port)

see how https://github.com/nim-lang/Nim/pull/14327 avoided this

Reply:
Yes, I based these tests off `tasynchttpserver.nim`. I looked at the linked PR, and it's not obvious that I can emulate that, because I need to use the `asyncHttpServer`, and I don't think it has a constructor that accepts a socket/FD. If you can suggest some code, I'm happy to try it and change it.

Reply:
resolved in later commits via https://github.com/nim-lang/Nim/pull/16678#issuecomment-758367414